### ISSUE TITLE
feature request: store thumbnails locally

### ISSUE THREAD

**Issue by winst0nsmith** [vauvenal5/yaga#158]
hi.

i looked didn't see anyone asking for this. i'd like better offline usability in the yaga app.

is it possible to have downloaded image thumbnails stored locally on the device, so that one could browse a photo library while offline. i'd also like to see full size images cached locally when viewed, but to have that cache size-limited and those cached images cleaned out based on last access date.

basically, i'd like to have yaga operate like ios photos app. this keeps the memory used on the device to a minimum, while allowing limited offline usability in the form of cached thumbnails and recent images.

thanks for the hard work so far tho!

**Comment by vauvenal5** [comment]
Hi!

Thumbnails of folders which were once browsed should be visible in offline mode. I have to take a look if something is broken in that regard. Or is your request about downloading all thumbnails which exist on the server?

Regarding your 2nd request with a max cache limit, I will keep it in mind but there are a few more important things on the table right now.

Thx for the feedback! :)

**Comment by winst0nsmith** [comment]
1. the entire library's thumbnails need downloaded automatically. they should persist when offline.
2. tapping an image thumbnail should download the full size (or perhaps a downscaled version of) that photo.
3. downloaded full-sized (or downscaled) photos should persist offline, until a threshold library size in GB is hit. at that point, the oldest full-size images should be removed first, with their thumbnails left behind.

this is the behavior apple's photos app follows, which allows for large libraries to be browsed, in their entirety, locally on the phone even when offline. i don't mean to sound like an apple fanboi, but their photos app is pretty damn slick. that should be the model. it is pretty magic. does anyone know if google photos behaves in a similar fashion? i have never used it.

i am no programmer, but it seems to me like you are close to this already. i wish i could help, like i say i think you've done a great job already, just throwing my wishlist at you...i really don't want to use either apple or google's photos apps for privacy reasons.
